src/components/Popup.js

```diff
@@ -6,32 +6,41 @@ class Popup {
 
   open() {
     this._popupElement.classList.add("modal__open");
-    document.addEventListener("keydown", this._handleEscClose);
+    this.addEventListeners();
   }
 
   close() {
     this._popupElement.classList.remove("modal__open");
-    document.removeEventListener("keydown", this._handleEscClose);
+    this.removeEventListeners();
   }
 
-  setEventListeners() {
-    this._popupElement.addEventListener("click", (evt) => {
-      if (evt.target === evt.currentTarget) {
-        this.close();
-      }
-    });
```

Comment:
Please, add the close icon listener into `setEventListeners` to set it only once